src/Adapter/Carrier/QueryHandler/GetAvailableCarriersHandler.php

```diff
@@ -101,33 +134,85 @@ public function handle(GetAvailableCarriers $query): GetCarriersResult
     }
 
     /**
-     * Compute intersection of carriers across all products.
+     * @param ProductQuantity[] $productQuantities
      *
-     * @return int[]
+     * @return Product[]
+     */
+    private function loadProducts(array $productQuantities): array
+    {
+        $products = [];
+        foreach ($productQuantities as $pq) {
+            $productId = $pq->getProductId();
+            $products[$productId->getValue()] = $this->productRepository->get(
+                $productId,
+                new ShopId($this->shopContext->getId())
+            );
+        }
+
+        return $products;
+    }
+
+    /**
+     * Ensures country is active.
+     */
+    private function assertCountryIsActive(CountryId $countryId): void
+    {
+        $country = $this->countryRepository->get($countryId);
+        if (!$country->active) {
+            throw new RuntimeException('Country is not active for address.');
+        }
+    }
+
+    /**
+     * Retrieves address from address ID.
+     */
+    private function getAddress(?AddressId $addressId): Address
+    {
+        if (!$addressId) {
+            throw new RuntimeException('No delivery address provided.');
+        }
+
+        return $this->addressRepository->get($addressId);
+    }
```

Comment:
If anything else than AddressId triggers an exception, why not letting PHP do its job on runtime?


```suggestion
    private function getAddress(AddressId $addressId): Address
    {
        return $this->addressRepository->get($addressId);
    }
```

A null value will trigger an Exception

Reply:
This check is no longer necessary, addressId was null before. Good point.